mlir-clang/Lib/clang-mlir.cc

```diff
@@ -6162,27 +6158,39 @@ static bool parseMLIR(const char *Argv0, std::vector<std::string> filenames,
 
   Argv.push_back("-emit-ast");
 
-  const unique_ptr<Compilation> compilation(
-      driver->BuildCompilation(llvm::ArrayRef<const char *>(Argv)));
+
+  llvm::SmallVector<const ArgStringList *, 4> CommandList;
+  ArgStringList InputCommandArgList;
+
+  unique_ptr<Compilation> compilation;
+
+  if (InputCommandArgs.empty()) {
+    compilation.reset(
+      std::move(driver->BuildCompilation(llvm::ArrayRef<const char *>(Argv))));
+
   JobList &Jobs = compilation->getJobs();
   if (Jobs.size() < 1)
     return false;
-
-  MLIRAction Act(fn, module);
-
   for (auto &job : Jobs) {
-    std::unique_ptr<CompilerInstance> Clang(new CompilerInstance());
-
     Command *cmd = cast<Command>(&job);
     if (strcmp(cmd->getCreator().getName(), "clang"))
       return false;
+      CommandList.push_back(&cmd->getArguments());
```

Comment:
```suggestion
    CommandList.push_back(&cmd->getArguments());
```

Removing the tabulation so we clearly see that this instruction is not in the scope of the ```if(strcmp)instruction```